src/app/compiler/compiler.js

```diff
@@ -281,12 +348,9 @@ function Compiler (handleImportCall) {
   }
 
   function updateInterface (data) {
-    for (var contract in data.contracts) {
-      var abi = JSON.parse(data.contracts[contract].interface)
-      abi = solcABI.update(truncateVersion(currentVersion), abi)
-      data.contracts[contract].interface = JSON.stringify(abi)
-    }
-
+    txHelper.visitContracts(data.contracts, (contract) => {
+      data.contracts[contract.file][contract.name].abi = solcABI.update(truncateVersion(currentVersion), contract.object.abi)
```

Comment:
What is `contract.object`? If that refers to `evm.bytecode.object` then this is wrong since `.abi` is not part of that,.

Reply:
`contract` ìs something like:
```
{
file,
name,
object
}
```
`object` does not refers to `evm.bytecode.object`

Reply:
`object` refers to `contracts[fileName][contractName]`